### PR TITLE
Create a First Public Working Draft for Data Integrity.

### PR DESCRIPTION
**Pull request by msporny**
This PR creates a First Public Working Draft for the Data Integrity specification.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/pull/67.html" title="Last updated on Oct 29, 2022, 1:26 PM UTC (74c54d9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-integrity/67/0d0605e...74c54d9.html" title="Last updated on Oct 29, 2022, 1:26 PM UTC (74c54d9)">Diff</a>

### CODE REVIEW

**Comment by iherman** [comment]
This is not my final review as I am out for a few days, just a remark after a rapid look... Now that we have a formal vocabulary document in the repository (just merged) wouldn't it be better to refer to that one from the start rather than to the CCG document?

File: index.html [hunk 1]
```diff
@@ -44,14 +44,16 @@
...
           }, {
             name: "Dave Longley", url: "https://digitalbazaar.com/",
             company: "Digital Bazaar", companyURL: "http://digitalbazaar.com/",
-            note: "2014-2022"
+            note: "2014-2022", w3cid: 48025
           }, {
             name: "Mike Prorock", url: "https://www.linkedin.com/in/mprorock",
-            company: "Mesur.io", companyURL: "https://mesur.io/"
+            company: "Mesur.io", companyURL: "https://mesur.io/",
+            w3cid: 0
```
Comment (msporny) [inline index.html]: @mprorock, we'll need your w3cid, which you can find from this page: https://www.w3.org/users/myprofile

**Comment by msporny** [comment]
> This is not my final review as I am out for a few days, just a remark after a rapid look... Now that we have a formal vocabulary document in the repository (just merged) wouldn't it be better to refer to that one from the start rather than to the CCG document?

Yes, I'll make that change in the main branch and rebase this one on top of main.

Fixed in 4a10325eb7f4142aa6172822cc9e897032e589a6.

**Review by dlongley — APPROVED** [review]

**Review by TallTed — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (iherman) [reply index.html]: It is 130636, @msporny @mprorock

Comment (msporny) [inline index.html]:
```suggestion
            w3cid: 130636
```

**Review by OR13 — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Fixed in bf73e97442a4fd82d580587db897ae427234e0be.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-10-26-vcwg#section1-3) on 2022-10-26

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.3. Create a First Public Working Draft for Data Integrity. (pr vc-data-integrity#67)

_See github pull request [vc-data-integrity#67](https://github.com/w3c/vc-data-integrity/pull/67)._




**Manu Sporny:** Moving over to the DI spec..  
… Last week we said we'd talk about doing FWPD this week and proposals and voting on them. This is a PR to do FPWD for the DI spec..  
… I believe it has much of what we agreed to in it, when we take up that proposal, Ivan mentioned that we need to ideally pick a publication date into the future like 2-3 weeks..  
… That we also need to specify the kind of doc we intend to publish and any editorial changes to do that and we need to specify that we want to use Echidna to auto-publish editor's drafts going forward..  

> *Brent Zundel:* and we need to choose the shortname.

**Manu Sporny:** The proposal needs all of those things in there..  
… I think that's it for updates on VCDM and DI specs..  

**Kristina Yasuda:** Fantastic, thank you do so much. For VC-JWT and JsonWebSignature2020...?.  

**Orie Steele:** For VC JWS2020, we have one with changes requested regarding the IANA context..  

</details>

**Review by iherman — CHANGES_REQUESTED** [review]
- SOTD:
  - This sounds a bit strange to me: "This is an experimental specification and is undergoing regular revisions. It is not fit for production deployment." If it is experimental, it should not be on a rec-track. I propose to simply remove this; the general boilerplate is good enough. 
- Introduction
   - shouldn't this section, or its subsections if appropriate, be marked as informative? 
- §2.1
   - the fonts used for domain, challenge, proofValue seem to be different...
- References:
  - (Same comment as for jws): the RCH WG is busy producing a FPWD for the RDF-Dataset-c14n. If they are quicker than we are, then the normative reference should use that one. If not, we should push (temporarily) that reference to the list of informal specs (the Director may object seeing a CG document as a normative reference). Also, If used, the reference should be  https://w3c-ccg.github.io/rdf-dataset-canonicalization/spec/index.html. cc @phila @peacekeeper 
  - Just to be precise, the security vocabulary has also my name as an editor (in the references) :-)
  - Although it is an informative reference, wouldn't URDNA2015 taken over by the RCH document, too?
  - Shouldn't the normative reference to VCDM go to https://www.w3.org/TR/vc-data-model-2.0/? 
  - ZCAP is a CCG draft, I do not think it should be used as a normative reference

**Comment by msporny** [comment]
> * SOTD:
>   * This sounds a bit strange to me: "This is an experimental specification and is undergoing regular revisions. It is not fit for production deployment." If it is experimental, it should not be on a rec-track. I propose to simply remove this; the general boilerplate is good enough.

Fixed in d22e29648d202cdf8f11792b7b84815056c71a0b.

> * Introduction
>   * shouldn't this section, or its subsections if appropriate, be marked as informative?
> * §2.1

Fixed in 09b4c6372aff8a0fa8bbfe8c45f78525fbff5019.

>   * the fonts used for domain, challenge, proofValue seem to be different...

Fixed in a9b9a247f5fc383ef344bfdf04cf5527bf366b3f.

> * References:
>   
>   * (Same comment as for jws): the RCH WG is busy producing a FPWD for the RDF-Dataset-c14n. If they are quicker than we are, then the normative reference should use that one. If not, we should push (temporarily) that reference to the list of informal specs (the Director may object seeing a CG document as a normative reference). Also, If used, the reference should be  https://w3c-ccg.github.io/rdf-dataset-canonicalization/spec/index.html. cc @phila @peacekeeper
>   * Just to be precise, the security vocabulary has also my name as an editor (in the references) :-)
>   * Although it is an informative reference, wouldn't URDNA2015 taken over by the RCH document, too?
>   * Shouldn't the normative reference to VCDM go to https://www.w3.org/TR/vc-data-model-2.0/?
>   * ZCAP is a CCG draft, I do not think it should be used as a normative reference

Fixed in 11629fd02e2061c1aa88cd897a2a9497900cc989.

**Comment by iherman** [comment]
The VCDM reference is still to the v1.0 Rec. It is o.k. to keep it that way for the FPWD but, eventually, this must be changed to v2.0; after all, these documents will be developed further hand in hand.

**Review by iherman — APPROVED** [review]

**Comment by msporny** [comment]
> The VCDM reference is still to the v1.0 Rec. It is o.k. to keep it that way for the FPWD but, eventually, this must be changed to v2.0; after all, these documents will be developed further hand in hand.

Updated to use v2.0 references in 74c54d9b90802ddd86c594035e5c2d0709e0f50c.

**Comment by msporny** [comment]
Editorial snapshot, multiple reviews, changes requested and made, no objections, merging.

**Comment by msporny** [comment]
@iherman The publishable FPWD is now ready at: https://w3c.github.io/vc-data-integrity/FPWD/2022-11-10/

I still have to pubrules/CSS/link check it and fix up any issues. Just generating the URL for the call today.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-11-02-vcwg#section1-4) on 2022-11-02

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.4.  (pr vc-data-integrity#67)

_See github pull request [vc-data-integrity#67](https://github.com/w3c/vc-data-integrity/pull/67)._




**Manu Sporny:** for the last week there was a PR open.  
… PR 67.  
… that has been done.  

> *Manu Sporny:* The FPWD ready draft for Data Integrity is here: [https://w3c.github.io/vc-data-integrity/FPWD/2022-11-10/](https://w3c.github.io/vc-data-integrity/FPWD/2022-11-10/).

**Manu Sporny:** ivan, chairs, that document is ready to be published by nov 10.  
… contains diagrams etc. in the sub directory, please take note of that, when publishing other docs that with the spec.  

> *Ivan Herman:* +1.

**Brent Zundel:** VC-JWT and JWS2020.  

</details>

**Pull request merged**